src/aiida/engine/processes/calcjobs/manager.py

```diff
@@ -144,11 +151,11 @@ async def _update_job_info(self) -> None:
 
             raise
         else:
-            for job_id, future in self._job_update_requests.items():
-                if not future.done():
-                    future.set_result(self._jobs_cache.get(job_id, None))
-        finally:
-            self._job_update_requests = {}
+            for job_id in self._polling_jobs:
+                future = self._job_update_requests.pop(job_id)
+                if future.done():
+                    continue
+                future.set_result(self._jobs_cache.get(job_id, None))
```

Comment:
I think this is safer, because I don't see any guarantee that job_ids in `self._polling_jobs` are necessarily present in `self._job_update_requests`, the the pop call can throw a KeyError.

```suggestion
            for job_id in set(self._job_update_requests.keys()):
                # Skip job_ids that were not polled yet (e.g. when the job update requests arrived in between `_get_jobs_from_scheduler` and `update_job_info`
                # However, ff the future is already done, we want to pop it from the dict regardless (hence the or condition)
                if not job_id in self._polling_jobs() or self._job_update:
                future = self._job_update_requests.pop(job_id)
                if not future.done():
                    future.set_result(self._jobs_cache.get(job_id, None))
```